.github/workflows/dummy_lambda.yml

```diff
@@ -1,12 +1,39 @@
 name: deploy-dummy-lambda
 on:
   push:
-    branches:
-      - lambda-deploy-action
+    branches: master
     paths:
       - 'dummy_lambda/**'
+  pull_request:
+    branches: master
 jobs:
```

Comment:
From what I understand, this is going to deploy the lambda function when a pull request is raised on master, any pull request, regardless of the changes. Is that intended?

Maybe you should add the path filter:
    paths:
      - 'dummy_lambda/**'
for pull_request as well...

Reply:
I don't think `dummy_lambda/**` will require here as deploying lambda will be done only when you push against the master. The `pull_request` , added here. As there will be only one yml file in future which will be used for deploy and test jobs. By adding `pull_request` we are checking here that tests are triggered whenever any PR raised against master. Since we don't want to deploy any code while creating PR but run only test the part of specifying path is skipped here. Let me know in case you have any further comments.

Reply:
@rahul-bhave, looks like the intent is to deploy the lambda when we push to master and not when a pull request is raised. However, from what I understand, both on "push" to master and "pull_request" to master we are going to trigger all the jobs specified. Which means, the job "build" is going to get triggered even for pull_requests. Just by not adding the path, I dont think we can run only "test" job and restrict the "build" job execution.

Reply:
@sravantit25 , If you can see my last commit `push build job` has not run here. So I think that validates the theory of running only test job while creating or updating "pull_request". The changes on the dummy-lambda, I am going to revert once the experiment of merging both deploy.yml and test.yml successful(My and Arun's conversation around this experimental change at the trello- `https://trello.com/c/arApwUbJ/10-merged-test-run-with-deploy-action`. Let me know your thoughts around this.

Reply:
'So I think that validates the theory of running only test job while creating or updating "pull_request"'

But which part of your code is restricting that? There should be some filter or condition in your job or in the workflow xml which is stopping the build job from being run. If you are sure that the build job will not run for pull requests, please go ahead and ignore my comments.